Version 05

WilliamDenniss opened the pull request.

WilliamDenniss committed:
Updated to address comments from Roshni.

selfissued commented:
Fine with me.  Of course, you'll want to also change the version number and add a corresponding history entry.

WilliamDenniss committed:
Updated with comments from Brian.
NB. the correction to the example was previously reported.

WilliamDenniss committed:
Added optional support for QR codes NFC by allowing clients to include the user_code on the verification URI as a hint.
Moved the non-browser discussion out of the main protocol flow.

WilliamDenniss committed:
Updated with feedback form Justin.
– Removed redundant response_type.
– Clarified expiry behavior.

selfissued commented on draft-ietf-oauth-device-flow.xml:
```diff
@@ -238,14 +238,11 @@
 	<t>The client initiates the flow by requesting a set of verification
 	codes from the authorization server by making an HTTP "POST" request
-	to the device endpoint.  The client constructs a request URI by
+	to the device authorization endpoint.  The client constructs a request URI by
```
I don't think this is a request URI - It's just a request posted at the device authorization endpoint.  Delete "URI".

WilliamDenniss replied:
Done. Clarified how the request is built.

WilliamDenniss committed:
Clarified how the request is built per Mike's suggestion.
Linebreaks for examples so they render in HTML correctly.
Added changelog for -05.

tigertoes commented on draft-ietf-oauth-device-flow.xml:
```diff
@@ -259,12 +257,14 @@
+  <t>
 	  <figure><artwork><![CDATA[
-   POST /token HTTP/1.1
+   POST /device_authorization HTTP/1.1
```
I'm intrigued by this particular change, as compared to its cousin ETSI standard ([ETSI TS 103 407](http://www.etsi.org/deliver/etsi_ts/103400_103499/103407/01.01.01_60/ts_103407v010101p.pdf)) the choice of URI paths appears to have evolved like so:

| Action | ETSI | IETF Draft 4 | This PR |
|--------|------|-------------|---------|
| Request user_code | POST /associate | POST /token | POST /device_authorization
| Device Auth | POST /token | POST /token | POST /token

As minor as it is, maybe having this endpoint match the ETSI equivalent might be a suitable idea?

WilliamDenniss replied:
The important point is that endpoints used in the "Device Authorization Request" and the "Device Access Token Request" are not the same endpoints. It was a bug in `-04` that the same name was used in both examples, which could give a misleading impression that they were the same endpoint.

I like the term "device_authorization" since it matches the terms we're using in the spec for that action, but if you want to make this proposal on the oauth IETF list to change it to "associate", we can do a poll and see what others think.

selfissued commented:
Agreed.  It's more important that the example path reflect the name usage in this spec than match example paths in a different spec.

WilliamDenniss committed:
Added text to clarify that unrecognised parameters MUST be ignored.
- Wording taken from RFC6749.

WilliamDenniss committed:
Fixed review comments from Mike Jones.

ve7jtb commented:
Just went through it again.  Looks good.

One thought is that in the next update in the non normative text it might be good to have an example of how a QR code could have the device_code as a query parameter.   I was just talking to someone who thought it needed to be entered separately.    No big deal but an example may help get the idea across.

ve7jtb commented:
Sorry I just saw that you did add that example.   
John B.
